lib/encoder.js

```diff
@@ -752,6 +755,31 @@ function defineInstanceMembers() {
     parts.unshift(this.getLengthBuffer(propCounter));
     return Buffer.concat(parts);
   };
+  this.encodeJson = function encodeJson(value){
+    let jsonString;
+    //Use a switch to handle other typeof checking in the future
+    switch(typeof value){
+      case 'string':
```

Comment:
The switch statement should be converted to an "if string else json" statement until there is a need for multiple type checking.